uniffi_bindgen/src/bindings/kotlin/gen_kotlin/mod.rs

```diff
@@ -511,6 +512,17 @@ mod filters {
         Ok(as_ct.as_codetype().literal(literal, ci))
     }
 
+    // Get the idiomatic Kotlin rendering of an individual enum variant's discriminant
+    pub fn variant_discr_literal(e: &Enum, index: &usize) -> Result<String, askama::Error> {
+        let literal = e.variant_discr(*index).expect("invalid index");
+        match literal {
+            // Kotlin doesn't convert between signed and unsigned by default
+            // so we'll need to make sure we define the type as unsigned
+            LiteralMetadata::UInt(v, _, _) => Ok(v.to_string() + "u"),
```

Comment:
[I think that even a signed repr is going to end up with a UInt here](https://github.com/mozilla/uniffi-rs/blob/26a2fef271e78e9be3cfa5385d96e2c130eaf219/uniffi_meta/src/reader.rs#L425) - ie, the type of each value can't really be trused, only the type on the enum's discr can.

Reply:
yes! That's what makes it slightly broken for signed right now on kotlin specifically (swift and python is fine) but not sure if we want to just have kotlin left out (if that's even possible to detect since it goes through the UInt route still).